sklearn/decomposition/incremental_pca.py

```diff
@@ -204,42 +208,45 @@ def partial_fit(self, X, y=None):
         else:
             self.n_components_ = self.n_components
 
-        if (self.components_ is not None) and (self.components_.shape[0]
-                                               != self.n_components_):
+        if (self.components_ is not None) and (self.components_.shape[0] !=
+                                               self.n_components_):
             raise ValueError("Number of input features has changed from %i "
                              "to %i between calls to partial_fit! Try "
-                             "setting n_components to a fixed value." % (
-                                 self.components_.shape[0], self.n_components_))
+                             "setting n_components to a fixed value." %
+                             (self.components_.shape[0], self.n_components_))
 
-        if self.components_ is None:
-            # This is the first pass through partial_fit
+        # This is the first partial_fit
+        if not hasattr(self, 'n_samples_seen_'):
             self.n_samples_seen_ = 0
-            col_var = X.var(axis=0)
-            col_mean = X.mean(axis=0)
+            self.mean_ = .0
+            self.var_ = .0
+
+        # Update stats - they are 0 if this is the fisrt step
+        col_mean, col_var, n_total_samples = \
+            _batch_mean_variance_update(X, last_mean=self.mean_,
+                                        last_variance=self.var_,
+                                        last_sample_count=self.n_samples_seen_)
+
+        # Whitening
+        if self.n_samples_seen_ == 0:
+            # If it is the first step, simply whiten X
             X -= col_mean
```

Comment:
`check_array` will do a copy but will try not to do several in a row (e.g. if `copy=True` and you convert the data dtype at the same time, it will only do a single copy).